src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
 package core.basesyntax;
 
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int STARTING_CAPACITY = 10;
+    private static final double INCREASE_FACTOR = 1.5;
+
+    private T[] frame;
+    private int size;
+
+    public ArrayList() {
+        frame = (T[]) new Object[STARTING_CAPACITY];
+        size = 0;
+    }
+
+    private void ensureCapacity() {
+        if (size == frame.length) {
+            int newCapacity = (int) (frame.length * INCREASE_FACTOR);
+            T[] newArray = (T[]) new Object[newCapacity];
+
+            for (int i = 0; i < size; i++) {
+                newArray[i] = frame[i];
+            }
+            frame = newArray;
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index: " + index);
+        }
+    }
+
+    /*IN MY PREVIOUS COMMIT #baf8c0e a for-each loop would not work in AddbyIndex.
+    for each isnt applicable to core.basesyntax.List<T>
+    I had to replace it with Iterator
+    Wyy???
```

Comment:
because our ArrayList doesn't implement Iterable
java.util.ArrayList implements List which extends Collection which extends Iterable
better remove iterator related changes and stick to the for loop